recipes-omnect/initrdscripts/omnect-os-initramfs/factory-reset

```diff
@@ -152,41 +122,70 @@ factory_reset_backup() {
 factory_reset_restore() {
     local path_name="$1"  # file or directory
     local ret=0
+    local destpath=$(dirname ${ROOTFS_DIR}${path_name})
+    local nested_mount=""
 
     msg "restore: ${path_name}"
 
-    if [ ! -e "${FACTORY_RESET_BACKUP_DIR}/${ROOTFS_DIR}/${path_name}" ]; then
+    if [ ! -e "${factory_reset_backup_dir}/${ROOTFS_DIR}/${path_name}" ]; then
         msg "->does not exist; ignore..."
         return 0
     fi
 
-    cd ${FACTORY_RESET_BACKUP_DIR}/${ROOTFS_DIR}
-        # do restore, keep permissions and ownership
-        run_cmd cp --parents -av "${path_name:1}" "${ROOTFS_DIR}"
+    while [[ "${destpath}" != "${ROOTFS_DIR}" ]]; do
+        mountpoint -q ${destpath} && nested_mount=${destpath/${ROOTFS_DIR}} && break
+        destpath=${destpath%\/*}
+    done
+
+    # do restore, keep permissions and ownership
+    if [ -n "${nested_mount}" ]; then
+        msg "nested mount: ${factory_reset_backup_dir}/${ROOTFS_DIR}${nested_mount}"
+        cd ${factory_reset_backup_dir}/${ROOTFS_DIR}${nested_mount}
+        path_name=${path_name/${nested_mount}}
+        run_cmd cp --parents -av "${path_name:1}" "${ROOTFS_DIR}${nested_mount}"
         ret=$?
-    cd -
+        cd - &>/dev/null
+    else
+        msg_error "restore: no mountpoint found in destination path: ${destpath}"
+        ret=66
+    fi
+
 
     sync
     return ${ret}
 }
 
 iterate_restore_list() {
-    local restore_list="$1"
-    local    do_backup="$2"
+    local    do_backup="${1}"
+    local restore_list="${@:2}"
+
     local ret=0
     local status=0
+    local context=""
 
-    restore_list=${restore_list//;/ }
-    for i in ${restore_list}; do
+    for i in ${restore_list[@]}; do
```

Comment:
afaics `restore_list` is no array here, so `[@]` can be omitted, the same applies below.

Reply:
it kinda is ... why should i omit it?

Reply:
ok, it doesn't hurt, right.
but what does `restore_list` make different from any other plain variable, e.g. like `i`, to use `[@]` here?
it's rather confusing than helping here imho, as it suggests we're dealing with an array here what is not the case.

Reply:
i'm confused. what makes you think, we're not dealing with an array here? see line 160.

Reply:
line 160 expands to a string containing all positional arguments starting with the 2nd and separated by spaces.
if you wanted to have an array, parentheses would be necessary like used in lines 250 and 252 for variable `preserve`, i.e. `local restore_list=("${@:2}")`

Reply:
Will possibly be revisited in a future PR.